### ISSUE TITLE
Entity/Hologram Ownership

### ISSUE THREAD

**Issue by awilliamson** [INPStarfall/Starfall#352]
Things such as 'holograms' aren't having their ownership set correctly. This can lead to some permissions issues for non-superadmin players in certain domains where the prop-protect goes a bit odd.

INP server allows things such as setColour on a N/A entity, SF doesn't complain, the PP ( Nadmod ) then informs the person executing said code that they now own that 'prop'.

It's been reported that on other servers SF will just give a permissions error when trying to set the colour of a hologram whose owner is N/A.

I've hunted through most of the spawnmenu, SENT spawning stuff from Garrysmod GitHub.

Here's some code:

``` lua
local ent = ents.Create( "prop_physics" )
-- Change Player index to whatever yours is
ent:SetPos( Player(4).PlayerTrace.HitPos )
ent:SetAngles( Angle( 0, 0, 0 ) )
ent:SetModel( "models/props_c17/FurnitureWashingmachine001a.mdl" )

ent:Spawn()

if IsValid( ent ) then
    -- https://github.com/garrynewman/garrysmod/blob/master/garrysmod/gamemodes/sandbox/gamemode/commands.lua#L613
    ent:SetCreator( Player( 4 ) )

    -- https://github.com/garrynewman/garrysmod/blob/master/garrysmod/lua/includes/extensions/entity.lua#L53
    ent:SetVar( "Player", Player( 4 ) )

    -- This was a guess based on the above, similar to the .Owner deprecation.
    ent:SetVar( "Owner", Player( 4 ) )
end

-- This is deprecated, but still supported. Even mentioned within official source
-- https://github.com/garrynewman/garrysmod/blob/master/garrysmod/lua/includes/extensions/entity.lua#L30
ent.Owner = Player( 4 )

-- Some undo stuff
undo.Create( "SomeProp" )
    undo.AddEntity( ent )
    undo.SetPlayer( Player( 4 ) )
undo.Finish()

-- THIS
-- This is the single line which turns owner from N/A -> PlayerName ( eg Radon )
-- This alone, none of the above things, even though I'm sure they 'help'
-- Tested in NADMOD singleplayer.
Player( 4 ):AddCleanup( "props", ent )
```

With the above code a prop_physics will spawn with the correct owner set, according to NADMOD mouseover HUD.

Using GCompute, Entity( 624 ) outputs thus:

``` lua
Entity (624)
{
    OnDieFunctions = { --[[ table: MEMORY ]] },
    Owner = player.GetByID (1) -- [[ STEAM_0:0:0, INP | Radon    ]],
    Player = player.GetByID (1) -- [[ STEAM_0:0:0, INP | Radon    ]],
    SPPOwner = player.GetByID (1) -- [[ STEAM_0:0:0, INP | Radon    ]],
    m_PlayerCreator = player.GetByID (1) -- [[ STEAM_0:0:0, INP | Radon    ]],
}
-- 5 total entries
```

Here:
- `Owner` comes from the `ent.Owner = Player( 4 )` command
- `Player` comes from the `ent:SetVar( "Player", Player( 4 ) )` command
- `m_PlayerCreator` comes from the `ent:SetCreator( Player( 4 ) )` command
- `SPPOwner` I presume comes from the `ent:AddCleanup( "props", ent )` command, which I again presume is caught by the PP system NADMOD.

Even when spawning props 'normally' through the spawn menu, the PP HUD overlay for Owner displays `N/A` briefly, before selecting the correct owner `INP | Radon`.

PS - Apologies for any typing mistakes, copy/paste crashes me at the moment, all code output is hand-written from reference.

**Comment by Divran** [comment]
http://ulyssesmod.net/archive/CPPI_v1-3.pdf

**Comment by awilliamson** [comment]
Thanks @Divran :)

**Comment by Xandaros** [comment]
The problem is, not all prop protection addons are CPPI compliant. It is a good idea to use the CPPI mechanisms if available, but maybe we should have a fallback for non-compliant addons.
(On the other hand... maybe they should just get a proper PP...)

**Comment by awilliamson** [comment]
Fixed by #360
